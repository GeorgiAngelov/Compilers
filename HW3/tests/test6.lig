function main(args : [[int]]) : int {
      // Print the command line arguments.

      var i : int;

      for (i = 0 to sizeof(args)) {
            print(args[i]);
      }

      return (0);
}
