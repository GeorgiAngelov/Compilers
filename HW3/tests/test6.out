(decls
   (decl-fun:ok (id main) (type-fun (params (typed-id (id args) (type-array (type-array (type-int))))) (type-int))
      (decls
         (decl-var:ok (id i) (type-int)))
      (stmts
         (for:ok (id:int i) (num:int 0) (fun-call:int (id sizeof) (exps (id:[[int]] args)))
            (stmts
               (exp-stmt:ok (fun-call:void (id print) (exps (array-idx:[int] (id:[[int]] args) (id:int i)))))))
         (return:ok (num:int 0)))))
